ingest/connection.py

```diff
@@ -0,0 +1,49 @@
+import abc
+import os
+
+from pymongo import MongoClient
+
+
+class Connection(metaclass=abc.ABCMeta):
+    """
+    The connetion interface that declares a client for a particular
+    client.
+    """
+    @property
+    @abc.abstractmethod
+    def client(self):
+        pass
+
+    @client.setter
+    @abc.abstractmethod
+    def client(self, x):
+        pass
+
+
+class MongoConnection(Connection):
+    """
+    A concrete class that defines a MongoDB client
+    """
+
+    def __init__(self):
+        self.host = os.environ['DATABASE_HOST']
+        self.user = os.environ['MONGODB_USERNAME']
+        self.password = os.environ['MONGODB_PASSWORD']
+        self.db_name = os.environ['DATABASE_NAME']
+
+    @property
+    def client(self):
+        return self._client
+
+    @client.setter
+    def client(self, x):
+        if os.environ.get('DATABASE_HOST') is not None:
+            self._client = MongoClient(
+                host,
+                username=user,
+                password=password,
+                authSource=db_name,
+                authMechanism='SCRAM-SHA-1',
+            )
+        else:
```

Comment:
The [comment from ingest pipeline](https://github.com/broadinstitute/scp-ingest-pipeline/blob/ac8786dcf246ac8d34cd8bbe867c981e16c92c73/ingest/ingest_pipeline.py#L118) says that check is ' Needed to run tests in CircleCI.  TODO: add mock, remove this'.

Reply:
I wrote the original connection code and comment, so I can offer some context.

> Needed to run tests in CircleCI.

Tests in [`test_ingest.py`](https://github.com/broadinstitute/scp-ingest-pipeline/blob/master/tests/test_ingest.py) fail in CI without something like this, as is.  The intent of those tests is to verify observable output using the same input interface as PAPI (i.e. using `ingest_pipeline.py` CLI arguments).  Such tests are a fast and accurate way to verify code works after lower-level implementation overhauls -- e.g. after a big refactoring like this.

> TODO: add mock, remove this

The `else` in question here exists because we do not use a database mock library for MongoDB.  Lacking such a mock DB means we need to mock an implementation detail (i.e. this DB reference, and a `load` function), against the intent of `test_ingest.py`.

It would be better to integrate a MongoDB mock, like [mongomock](https://github.com/mongomock/mongomock).  We [had a similar mock for Firestore](https://github.com/broadinstitute/scp-ingest-pipeline/pull/26), but never got around to the same infrastructure for MongoDB after pivoting.

SCP-2000 tracks that test infrastructure goal.  Refining the comment to:

> // TODO (SCP-2000): Integrate MongoDB emulator to test_ingest.py, then remove this

would better convey that goal.